tasks/totient-function/src/main.rs

```diff
@@ -0,0 +1,27 @@
+use num::integer::gcd;
+
+fn main() {
+    // Compute the totient of the first 25 natural integers
+    println!("N\t phi(n)\t Prime");
+    for n in 1..26 {
+        let phi_n = phi(n);
+        println!("{}\t {}\t {:?}", n, phi_n, phi_n == n - 1);
+    }
+
+    // Compute the number of prime numbers for various steps
+    [1, 100, 1000, 10000, 100000]
+        .windows(2)
+        .scan(0, |acc, tuple| {
```

Comment:
It might be cleaner here to destructure the tuple in the argument, like `|acc, (lower, upper)|`

Reply:
It would be nice to have it but when I've tried to do like 
```
    [1, 100, 1000, 10000, 100000]
        .windows(2)
        .scan(0, |acc, [lower, upper]| {
            *acc += (lower..=upper).filter(is_prime).count();
            Some((upper, *acc))
        })
        .for_each(|x| println!("Until {}: {} prime numbers", x.0, x.1));
```
I got something like
```

error[E0599]: no method named `filter` found for type `std::ops::RangeInclusive<&{integer}>` in the current scope
  --> tasks/totient-function/src/main.rs:15:37
   |
15 |             *acc += (lower..=upper).filter(is_prime).count();
   |                                     ^^^^^^
   |
   = note: the method `filter` exists but the following trait bounds were not satisfied:
           `&mut std::ops::RangeInclusive<&{integer}> : std::iter::Iterator`
           `std::ops::RangeInclusive<&{integer}> : std::iter::Iterator`
```
in return; going for `[&lower, &upper]` instead yield `expected integer, found reference` errors.

If you have any idea of how to do this cleanly, that'd be nice :)

Reply:
Oh I see, it's a slice, not a tuple. In that case, maybe this is the best you can do:

```rust
        .scan(0, |acc, window| {
            let (lower, upper) = (window[0], window[1]);
            *acc += (lower..=upper).filter(is_prime).count();
            Some((upper, *acc))
        })
```

Reply:
Should be done :)